sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs

```diff
@@ -288,14 +288,12 @@ public TimeSpan TimeToLive
         /// </remarks>
         public string CorrelationId
         {
-            get => _correlationIdSet ? AmqpMessage.Properties.CorrelationId?.ToString() : AmqpMessage.Properties.CorrelationId.ToString();
+            get => AmqpMessage.Properties.CorrelationId?.ToString();
```

Comment:
This will force allocation of the `Properties` collection if nothing in it was set - is that a concern?    Should we consider doing a section check before accessing?

```csharp
if (AmqpMessage.HasSection(AmqpMessageSection.Properties))
{
    return AmqpMessage.Properties.CorrelationId?.ToString();
}

return null;
```

Reply:
Currently, we always allocate the Properties section in the AMQP message anyway when converting from SB Message to AMQP message, so it wouldn't really matter what we do here. I just checked with Track 1 and the behavior is the same there. I'm not sure if this sends any extra bytes, but I haven't seen any issues around this. If we were to change this behavior we'd need to change it all the way down to the AmqpMessageConverter, but given this is the same behavior as T1, and no complaints, I would prefer to leave it.